### PR TITLE
TW-131: send one image without caption

### PR DESCRIPTION
**Pull request by sherlockvn**
https://github.com/linagora/twake-on-matrix/assets/43041967/687a656b-96bd-48ee-b2d1-ccac7c5cda19

https://github.com/linagora/twake-on-matrix/assets/43041967/6119bb60-4734-4c1a-af17-cec4e3da7528

### CODE REVIEW

**Comment by hoangdat** [comment]
I think you still miss the progress icon in the center of image
 image

File: lib/pages/chat/chat.dart [hunk 1]
```diff
@@ -397,28 +398,25 @@ class ChatController extends State<Chat> {
...
-  }
+  // Future<void> sendImages() async {
+  //   final selectedAssets = imagePickerController.sortedSelectedAssets;
+  //   for (final entity in selectedAssets) {
+  //     await sendImage();
+  //   }
+
+  //   imagePickerController.clearAssetCounter();
+  //   numberSelectedImagesNotifier.value = 0;
+  // }
```
Comment (nqhhdev) [inline lib/pages/chat/chat.dart:419]: Remove it?

File: lib/pages/chat/chat.dart [hunk 2]
```diff
@@ -397,28 +398,25 @@ class ChatController extends State<Chat> {
...
-      });
+  void sendImage() {
+    final assetEntity = imagePickerController.selectedAssets.first;
+    final sendImageInteractor = getIt.get<SendImageInteractor>();
+    if (assetEntity.asset.type == AssetType.image) {
+      sendImageInteractor.execute(room: room!, entity: assetEntity.asset);
+      imagePickerController.clearAssetCounter();
+    numberSelectedImagesNotifier.value = 0;
     }
   }
```
Comment (nqhhdev) [inline lib/pages/chat/chat.dart:409]:
```suggestion
  void sendImage() {
    final assetEntity = imagePickerController.selectedAssets.first;
    final sendImageInteractor = getIt.get<SendImageInteractor>();
    if (assetEntity.asset.type == AssetType.image) {
      sendImageInteractor.execute(room: room!, entity: assetEntity.asset);
      imagePickerController.clearAssetCounter();
      numberSelectedImagesNotifier.value = 0;
    }
  }
```

File: lib/pages/chat/events/message_content.dart [hunk 3]
```diff
@@ -112,6 +114,20 @@ class MessageContent extends StatelessWidget {
       case EventTypes.Sticker:
         switch (event.messageType) {
           case MessageTypes.Image:
+            if (event.status == EventStatus.error && event.messageType == MessageTypes.Image) {
+              return SizedBox(
+                width: MessageContentStyle.imageBubbleWidth,
+                height: MessageContentStyle.imageBubbleHeight,
+                child: const Center(
+                  child: Icon(Icons.error, color: Colors.red,),
```
Comment (nqhhdev) [inline lib/pages/chat/events/message_content.dart:122]:
```suggestion
                  child: Icon(Icons.error, color: Colors.red),
```

File: lib/utils/matrix_sdk_extensions/event_extension.dart [hunk 4]
```diff
@@ -46,4 +46,29 @@ extension LocalizedBody on Event {
...
+      if (status == EventStatus.synced || status == EventStatus.sent) {
+          if (unsigned?.containsKey('transaction_id') == true) {
+            final transactionId = unsigned!['transaction_id'];
+            data = room.sendingFilePlaceholders[transactionId]?.bytes;
+            return data;
+          }
+        }
+    }
+    return null;
+  }
```
Comment (nqhhdev) [inline lib/utils/matrix_sdk_extensions/event_extension.dart:62]: IMO :

```suggestion
  Uint8List? _getPlaceHolderMatrixFile(Event event) {
    if (room.sendingFilePlaceholders.isNotEmpty) {
      if (status == EventStatus.synced || status == EventStatus.sent) {
          if (unsigned?.containsKey('transaction_id') == true) {
            final transactionId = unsigned!['transaction_id'];
            final placeHolderMatrixFile = room.sendingFilePlaceholders[transactionId]?.bytes;
            return placeHolderMatrixFile;
          }
        }
    }
    return null;
  }
```

**Comment by nqhhdev** [comment]
Pls demo some case : 
1. If sent error => Retry send an image
2. Please send an image with size = `maxSendingImage`
3. Please send an image with sized > `maxSendingImage`

**Comment by sherlockvn** [comment]
> I think you still miss the progress icon in the center of image image

Currently, there is no progress for upload image, so we can only fake it

File: lib/pages/chat/chat.dart (hunk 1 above)
Comment (sherlockvn) [reply lib/pages/chat/chat.dart:419]: I think i will use it for next ticket about send multiple images

**Comment by sherlockvn** [comment]
> Pls demo some case :
> 
> 1. If sent error => Retry send an image
> 2. Please send an image with size = `maxSendingImage`
> 3. Please send an image with sized > `maxSendingImage`

It's the error case for sure

File: lib/domain/usecase/send_image_interactor.dart [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+class SendImageInteractor {
+  Future<void> execute({
+    required Room room,
+    required AssetEntity entity,
+    String? txId,
+    Event? inReplyTo,
+    String? editEventId,
+    int? shrinkImageMaxDimension,
+    MatrixImageFile? thumbnail,
```
Comment (hoangdat) [inline lib/domain/usecase/send_image_interactor.dart:15]: how about create thumbnail inside interactor too?

File: lib/presentation/extensions/room_extension.dart [hunk 6]
```diff
@@ -0,0 +1,230 @@
+import 'package:matrix/matrix.dart';
+import 'package:matrix/src/utils/file_send_request_credentials.dart';
+
+extension SendImage on Room {
+
+  static const maxSendingImage = 20;
+
+  Future<String?> sendImageFileEvent(
+    MatrixFile file, {
```
Comment (hoangdat) [inline lib/presentation/extensions/room_extension.dart:9]: how about push the fake event before creating file?

File: lib/presentation/extensions/room_extension.dart [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+    String? txid,
+    Event? inReplyTo,
+    String? editEventId,
+    int? shrinkImageMaxDimension,
+    MatrixImageFile? thumbnail,
```
Comment (hoangdat) [inline lib/presentation/extensions/room_extension.dart:14]: same here

File: lib/di/send_image/send_image_di.dart [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+import 'package:fluffychat/domain/usecase/send_image_interactor.dart';
+import 'package:get_it/get_it.dart';
+
+class SendImageDi extends BaseDI {
+  @override
+  String get scopeName => "Send image";
+
+  @override
+  void setUp(GetIt get) {
+    get.registerFactory<SendImageInteractor>(() => SendImageInteractor());
```
Comment (hoangdat) [inline lib/di/send_image/send_image_di.dart:11]: we need it factory? I think `singleton` inside this scope is OK.

File: lib/presentation/extensions/room_extension.dart (hunk 6 above)
Comment (sherlockvn) [reply lib/presentation/extensions/room_extension.dart:9]: But, the placeholder is matrix file, and also the creating file is very fast as i test, so i think no problem at all, maybe when sending multiple images, create files may be slow

**Review by hoangdat — APPROVED** [review]

**Pull request merged**